Library/Homebrew/test/cmd/list_spec.rb

```diff
@@ -1,12 +1,10 @@
 # frozen_string_literal: true
 
-require "cmd/shared_examples/args_parse"
+require "cmd/list"
 
-RSpec.describe "brew list" do
+RSpec.describe Homebrew::Cmd::List do
   let(:formulae) { %w[bar foo qux] }
 
-  it_behaves_like "parseable arguments"
```

Comment:
This doesn't play nicely with the new interface, but i don't think it's an especially useful test (it validates that `parse` can be called on the output of the command method, which has already called `parse` in nearly all cases). I think there are better tests in `abstract_command_spec`, including passing switches/flags.

Reply:
I agree that it's not that useful. I would expect that using Sorbet here covers most of the same logic (all hail the type checker!). It might be useful to test that the contents of the `cmd_args` block is valid. That should be covered by integration tests though so I don't think we need to add anything extra. Maybe just something to keep in mind if a command doesn't have an integration test. I can't remember if they all have integration tests or not.

Reply:
@dduugg I disagree, this has caught legitimate issues for me in some tests that other tests have not (or other tests have been sufficiently opaque in their errors whereas this was very clear). Similarly, there's a few commands that can't really have good/any integration tests and this helps there too.

Would like to have this stick around in some form if possible! If not possible: let's talk/think about how we could have a test that does similar.

Reply:
@MikeMcQuaid Thanks for the feedback. Let me know if the latest commit offers a viable replacement.

Methodology:

The `parseable arguments` test calls `#{command_name.tr("-", "_")}_args"` on `Homebrew`, in order to validate that the returned value responds to `parse` (I'm not sure why it does this, rather than check that it's an instance of `Homebrew::CLI::Parser.new`, but either way this would be better solved via `sig` enforcement, IMO)

In the new approach, `CLI::Parser` is called within the command initializer, where the parsing also happens. So, it would seem that validating that one can call `.new` on a command would be a stronger test than what was here before. I've also added the ability to pass a custom argv (since e.g. `prof` requires at least one argument).